pkg/ccl/testccl/sqlccl/sqlccl_test: TestExplainGist failed

pkg/ccl/testccl/sqlccl/sqlccl_test.TestExplainGist [failed](https://mesolite.cluster.engflow.com/invocations/default/06721d97-8731-4f7a-a454-3d3f6fa4b55d?testReportRun=3&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3FsY2NsOnNxbGNjbF90ZXN0) on release-25.4 @ [095068c58091c9152c33c223af10ca71165c7ddd](https://github.com/cockroachdb/cockroach/commits/095068c58091c9152c33c223af10ca71165c7ddd):


```
        google.golang.org/grpc/internal/transport.(*http2Server).keepalive(0xc002a29040)
        	external/org_golang_google_grpc/internal/transport/http2_server.go:1150 +0x1f3
        created by google.golang.org/grpc/internal/transport.NewServerTransport in goroutine 29452
        	external/org_golang_google_grpc/internal/transport/http2_server.go:339 +0x1a8e
        
        goroutine 29627 [select]:
        github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.(*metricsPoller).Resume(0xc0062a3720?, {0x8791538, 0xc00923f050}, {0x739ca40, 0xc0193f3720})
        	pkg/jobs/metricspoller/poller.go:68 +0x25a
        github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc004a4e180?, 0x0?, 0xc0152ef928, {0x878f490?, 0xc0193f36f0?}, {0x8791538?, 0xc00923f050?}, {0x739ca40?, 0xc0193f3720?})
        	pkg/jobs/registry.go:1645 +0xb2
        github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc003b7e808, {0x8791500, 0xc0062a3720}, {0x739ca40, 0xc0193f3720}, {0x878f490, 0xc0193f36f0}, 0xc004a4e180, {0xc0089d0372, 0x7}, ...)
        	pkg/jobs/registry.go:1646 +0xba5
        github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc003b7e808, {0x8791500, 0xc0062a3720}, {0x878f490, 0xc0193f36f0}, 0xc004a4e180, {0xc0089d0372, 0x7}, {0xc00243aa00, 0x7})
        	pkg/jobs/adopt.go:414 +0x545
        github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1({0x8791538?, 0xc00923ecf0?})
        	pkg/jobs/adopt.go:267 +0x13c
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x8791538, 0xc00923ecf0})
        	pkg/util/stop/stopper.go:440 +0x8a
        created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 29603
        	pkg/util/stop/stopper.go:438 +0x137
        
        goroutine 29267 [select]:
        github.com/cockroachdb/cockroach/pkg/sql.(*mvccStatisticsUpdateJob).runTenantGlobalMetricsExporter(0xc00ea13428, {0x8791538, 0xc00e2e9c50}, {0x87fc110, 0xc019f79790})
        	pkg/sql/mvcc_statistics_update_job.go:136 +0x1ea
        github.com/cockroachdb/cockroach/pkg/sql.(*mvccStatisticsUpdateJob).Resume(0xc00ea13428, {0x8791538, 0xc00e2e9c50}, {0x739ca40, 0xc019f79790})
        	pkg/sql/mvcc_statistics_update_job.go:74 +0x10a
        github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc009347c80?, 0x0?, 0xc01e5b5928, {0x876b3a0?, 0xc00ea13428?}, {0x8791538?, 0xc00e2e9c50?}, {0x739ca40?, 0xc019f79790?})
        	pkg/jobs/registry.go:1645 +0xb2
        github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc0056bd008, {0x8791500, 0xc0019fac80}, {0x739ca40, 0xc019f79790}, {0x876b3a0, 0xc00ea13428}, 0xc009347c80, {0xc00243a039, 0x7}, ...)
        	pkg/jobs/registry.go:1646 +0xba5
        github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc0056bd008, {0x8791500, 0xc0019fac80}, {0x876b3a0, 0xc00ea13428}, 0xc009347c80, {0xc00243a039, 0x7}, {0xc0023b8b76, 0x7})
        	pkg/jobs/adopt.go:414 +0x545
        github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1({0x8791538?, 0xc009c3b620?})
        	pkg/jobs/adopt.go:267 +0x13c
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x8791538, 0xc009c3b620})
        	pkg/util/stop/stopper.go:440 +0x8a
        created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 29219
        	pkg/util/stop/stopper.go:438 +0x137
    explain_test.go:311: stmt wasn't canceled by statement_timeout of 0.1s - ran at least for 1m: UPDATE
        	defaultdb.public.tab_192 AS tab_831
        SET
        	_int2 = 13319:::INT8
        FROM
        	defaultdb.public.tab_7 AS tab_832,
        	defaultdb.public.tab_7 AS tab_833,
        	defaultdb.public.tab_192 AS tab_834,
        	defaultdb.public.tab_7 AS tab_835,
        	defaultdb.public.tab_7 AS tab_836,
        	defaultdb.public.tab_192 AS tab_837
--- FAIL: TestExplainGist/main (88.43s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExplainGist.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-56335

--- a/pkg/sql/colexec/colexecjoin/crossjoiner.go
+++ b/pkg/sql/colexec/colexecjoin/crossjoiner.go
@@ -175,6 +175,7 @@ func (c *crossJoiner) consumeRightInput(ctx context.Context) {
 	// Consume the right input if necessary.
 	if needRightTuples || needOnlyNumRightTuples {
 		for {
+			c.cancelChecker.CheckEveryCall()
 			batch := c.InputTwo.Next()
 			if needRightTuples {
 				c.rightTuples.Enqueue(ctx, batch)

